From ea910a94ae684a42617e97496f4aa9bc630d323f Mon Sep 17 00:00:00 2001
From: Jonathon Penix <jpenix@quicinc.com>
Date: Tue, 25 Nov 2025 14:14:12 -0800
Subject: [PATCH] Remove hardcoded RISC-V multilibs

This might not be necessary for our final configs, but for the temp
ones I'm experimenting with, this messes up the library include paths
in a way that isn't consistent with Arm/AArch64.

We don't want these multilibs anyway, so just remove it for now.

Signed-off-by: Jonathon Penix <jpenix@quicinc.com>
---
 clang/lib/Driver/ToolChains/BareMetal.cpp | 69 -----------------------
 1 file changed, 69 deletions(-)

diff --git a/clang/lib/Driver/ToolChains/BareMetal.cpp b/clang/lib/Driver/ToolChains/BareMetal.cpp
index e128c31cd567..69c294c29483 100644
--- a/clang/lib/Driver/ToolChains/BareMetal.cpp
+++ b/clang/lib/Driver/ToolChains/BareMetal.cpp
@@ -51,70 +51,6 @@ static bool isPPCBareMetal(const llvm::Triple &Triple) {
          Triple.getEnvironment() == llvm::Triple::EABI;
 }
 
-static bool findRISCVMultilibs(const Driver &D,
-                               const llvm::Triple &TargetTriple,
-                               const ArgList &Args, DetectedMultilibs &Result) {
-  Multilib::flags_list Flags;
-  std::string Arch = riscv::getRISCVArch(Args, TargetTriple);
-  StringRef Abi = tools::riscv::getRISCVABI(Args, TargetTriple);
-
-  if (TargetTriple.isRISCV64()) {
-    MultilibBuilder Imac =
-        MultilibBuilder().flag("-march=rv64imac").flag("-mabi=lp64");
-    MultilibBuilder Imafdc = MultilibBuilder("/rv64imafdc/lp64d")
-                                 .flag("-march=rv64imafdc")
-                                 .flag("-mabi=lp64d");
-
-    // Multilib reuse
-    bool UseImafdc =
-        (Arch == "rv64imafdc") || (Arch == "rv64gc"); // gc => imafdc
-
-    addMultilibFlag((Arch == "rv64imac"), "-march=rv64imac", Flags);
-    addMultilibFlag(UseImafdc, "-march=rv64imafdc", Flags);
-    addMultilibFlag(Abi == "lp64", "-mabi=lp64", Flags);
-    addMultilibFlag(Abi == "lp64d", "-mabi=lp64d", Flags);
-
-    Result.Multilibs =
-        MultilibSetBuilder().Either(Imac, Imafdc).makeMultilibSet();
-    return Result.Multilibs.select(D, Flags, Result.SelectedMultilibs);
-  }
-  if (TargetTriple.isRISCV32()) {
-    MultilibBuilder Imac =
-        MultilibBuilder().flag("-march=rv32imac").flag("-mabi=ilp32");
-    MultilibBuilder I = MultilibBuilder("/rv32i/ilp32")
-                            .flag("-march=rv32i")
-                            .flag("-mabi=ilp32");
-    MultilibBuilder Im = MultilibBuilder("/rv32im/ilp32")
-                             .flag("-march=rv32im")
-                             .flag("-mabi=ilp32");
-    MultilibBuilder Iac = MultilibBuilder("/rv32iac/ilp32")
-                              .flag("-march=rv32iac")
-                              .flag("-mabi=ilp32");
-    MultilibBuilder Imafc = MultilibBuilder("/rv32imafc/ilp32f")
-                                .flag("-march=rv32imafc")
-                                .flag("-mabi=ilp32f");
-
-    // Multilib reuse
-    bool UseI = (Arch == "rv32i") || (Arch == "rv32ic");    // ic => i
-    bool UseIm = (Arch == "rv32im") || (Arch == "rv32imc"); // imc => im
-    bool UseImafc = (Arch == "rv32imafc") || (Arch == "rv32imafdc") ||
-                    (Arch == "rv32gc"); // imafdc,gc => imafc
-
-    addMultilibFlag(UseI, "-march=rv32i", Flags);
-    addMultilibFlag(UseIm, "-march=rv32im", Flags);
-    addMultilibFlag((Arch == "rv32iac"), "-march=rv32iac", Flags);
-    addMultilibFlag((Arch == "rv32imac"), "-march=rv32imac", Flags);
-    addMultilibFlag(UseImafc, "-march=rv32imafc", Flags);
-    addMultilibFlag(Abi == "ilp32", "-mabi=ilp32", Flags);
-    addMultilibFlag(Abi == "ilp32f", "-mabi=ilp32f", Flags);
-
-    Result.Multilibs =
-        MultilibSetBuilder().Either(I, Im, Iac, Imac, Imafc).makeMultilibSet();
-    return Result.Multilibs.select(D, Flags, Result.SelectedMultilibs);
-  }
-  return false;
-}
-
 static std::string computeClangRuntimesSysRoot(const Driver &D,
                                                bool IncludeTriple) {
   if (!D.SysRoot.empty())
@@ -340,11 +276,6 @@ void BareMetal::findMultilibs(const Driver &D, const llvm::Triple &Triple,
     Multilibs = Result.Multilibs;
     MultilibMacroDefines.append(CustomFlagMacroDefines.begin(),
                                 CustomFlagMacroDefines.end());
-  } else if (isRISCVBareMetal(Triple) && !detectGCCToolchainAdjacent(D)) {
-    if (findRISCVMultilibs(D, Triple, Args, Result)) {
-      SelectedMultilibs = Result.SelectedMultilibs;
-      Multilibs = Result.Multilibs;
-    }
   }
 }
 
-- 
2.17.1

